### PR TITLE
Correction in oval part ensure_gpgcheck_globally_activated

### PR DESCRIPTION
**Pull request by rumch-se**
#### Description:

- _Correction in OVAL part of the rule ensure_gpgcheck_globally_activated_

#### Rationale:

- According to DISA STIG for SUSE check should include that "gpgcheck" is not set to "0", "off", "no", or "false", because  this is a finding.

### CODE REVIEW

File: linux_os/guide/system/software/updating/ensure_gpgcheck_globally_activated/oval/shared.xml [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
   </definition>
+
   <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="check value of gpgcheck in {{{ pkg_manager_config_file }}}" id="test_ensure_gpgcheck_globally_activated" version="1">
     <ind:object object_ref="object_ensure_gpgcheck_globally_activated" />
   </ind:textfilecontent54_test>
+  
   <ind:textfilecontent54_object id="object_ensure_gpgcheck_globally_activated" comment="gpgcheck set in {{{ pkg_manager_config_file }}}" version="1">
     <ind:filepath>{{{ pkg_manager_config_file }}}</ind:filepath>
-    <ind:pattern operation="pattern match">^\s*gpgcheck\s*=\s*1\s*$</ind:pattern>
+    <ind:pattern operation="pattern match">^\s*gpgcheck\s*=\s*1s*$</ind:pattern>
```
Comment (jan-cerny) [inline linux_os/guide/system/software/updating/ensure_gpgcheck_globally_activated/oval/shared.xml:25]: This change looks wrong, there shouldn't be a literal `s`.

File: linux_os/guide/system/software/updating/ensure_gpgcheck_globally_activated/oval/shared.xml [hunk 2]
```diff
@@ -1,18 +1,42 @@
...
     {{{ oval_metadata("The gpgcheck option should be used to ensure that checking
       of an RPM package's signature always occurs prior to its
       installation.") }}}
-    <criteria operator="AND">
-     <criterion comment="check value of gpgcheck in {{{ pkg_manager_config_file }}}" test_ref="test_ensure_gpgcheck_globally_activated" />
-    </criteria>
+    {{% if 'sle' in product %}} 
+      <criteria operator="AND">
+        <criterion comment="check value of gpgcheck in {{{ pkg_manager_config_file }}}" test_ref="test_ensure_gpgcheck_globally_activated" />
+        <criterion comment="no value 0_off_no_false set in {{{ pkg_manager_config_file }}}" test_ref="test_ensure_gpgcheck_globally_no_deactivated" />
```
Comment (jan-cerny) [inline linux_os/guide/system/software/updating/ensure_gpgcheck_globally_activated/oval/shared.xml:10]: Can we have a test scenario where the first condition is met  but the second isn't met?

**Comment by rumch-se** [comment]
Hello @jan-cerny
I added 3 additional tests and fixed the issue with regexpr. 
Have a nice day
Rumen

**Review by jan-cerny — APPROVED** [review]
The fail of Automatus job is caused by the fact that we test a RHEL 8 data stream on a Fedora container but Fedora differs from RHEL in the path of the package manager config file (/ect/dnf/dnf.conf vs /etc/yum.conf). If we execute the test locally against a VM back end, everything passes:

```
jcerny@fedora:~/work/git/scap-security-guide (pr/11709)$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel9 ensure_gpgcheck_globally_activated
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2024-03-15-1025/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_ensure_gpgcheck_globally_activated
INFO - Script comment.fail.sh using profile (all) OK
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script line_not_there.fail.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK
INFO - Script wrong_value_false.fail.sh using profile (all) OK
INFO - Script wrong_value_no.fail.sh using profile (all) OK
INFO - Script wrong_value_off.fail.sh using profile (all) OK
jcerny@fedora:~/work/git/scap-security-guide (pr/11709)$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel9 --remediate-using ansible ensure_gpgcheck_globally_activated
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2024-03-15-1028/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_ensure_gpgcheck_globally_activated
INFO - Script comment.fail.sh using profile (all) OK
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script line_not_there.fail.sh using profile (all) OK
INFO - Script wrong_value.fail.sh using profile (all) OK
INFO - Script wrong_value_false.fail.sh using profile (all) OK
INFO - Script wrong_value_no.fail.sh using profile (all) OK
INFO - Script wrong_value_off.fail.sh using profile (all) OK
```

**Pull request merged**